server/controllers/referral.controllers.js

```diff
@@ -0,0 +1,70 @@
+import {
+  getAllUserReferrals,
+  updateReferralToRewarded,
+  getReferralByEmail,
+  sendReferralInvite,
+} from '../services/referral.service';
+import httpStatus from '../helpers/httpStatus';
+import EMAIL_CONTENT from '../../mailer';
+import { sendMail } from '../services/mailer.service';
+
+const ReferralController = {
+  getAllUserReferrals(req, res, next) {
+    const referrerId = req.params.id;
+    getAllUserReferrals(referrerId)
+      .then((referrals) => {
+        res.status(httpStatus.OK).json({ success: true, referrals });
+      })
+      .catch((error) => next(error));
+  },
+
+  getReferrals(req, res, next) {
```

Comment:
I am not sure of your thought process for this but I think it is serving the same purpose as the method above. Lets make the name more descriptive as this is a better way to get the user referral:
```suggestion
  getUserReferrals(req, res, next) {
```

Reply:
this method gets the referrals a user has sent. that's why the user id is gotten from the token. the method above does the same thing, but it's used by an admin to ge a user's referrals by passing their id in the url